go/pkg/apis/recognizer/devfile_recognizer.go

```diff
@@ -0,0 +1,81 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc.
+ ******************************************************************************/
+package recognizer
+
+import (
+	"errors"
+	"strings"
+
+	"github.com/redhat-developer/alizer/pkg/apis/language"
+)
+
+type DevFileType struct {
+	Name        string
+	Language    string
+	ProjectType string
+	Tags        []string
+}
+
+func SelectDevFileFromTypes(path string, devFileTypes []DevFileType) (DevFileType, error) {
+	languages, err := Analyze(path)
+	if err != nil {
+		return DevFileType{}, err
+	}
+	for _, language := range languages {
+		devfile, err := selectDevFileByLanguage(language, devFileTypes)
+		if err == nil {
```

Comment:
What if err is not nil; seems error is ignored

Reply:
Yeah because `selectDevFileByLanguage` only return an error when no devfile is found for the current language. Maybe i can return nil without any error :thinking: . BTW i keep looping because if first language doesn't have a devfile, maybe the second could have one.